[Safe Browsing] Consolidate SSBLookupContext usage in a single utility file

whsieh opened the pull request:
#### e8f0c2db123214bdb08f0bb31681f6a63b161369
<pre>
[Safe Browsing] Consolidate SSBLookupContext usage in a single utility file
<a href="https://bugs.webkit.org/show_bug.cgi?id=303316">https://bugs.webkit.org/show_bug.cgi?id=303316</a>
<a href="https://rdar.apple.com/165620397">rdar://165620397</a>

Reviewed by Abrar Rahman Protyasha.

Wrap code to call into SafariSafeBrowsing for lookup in C++ helper functions, in a new utility file
and namespace (`SafeBrowsingUtilities::lookUp`, etc.). No change in behavior.

* Source/WebKit/Platform/spi/Cocoa/SafeBrowsingSPI.h:
* Source/WebKit/SourcesCocoa.txt:

Add a new file, `SafeBrowsingUtilities.{h|mm}`, and move soft-linking to the source, since we now
only need to call directly into SSB classes in this file.

* Source/WebKit/UIProcess/Cocoa/BrowsingWarningCocoa.mm:
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.h: Removed.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingUtilities.h: Renamed from Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.mm.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingUtilities.mm: Added.
(WebKit::SafeBrowsingUtilities::canLookUp):
(WebKit::SafeBrowsingUtilities::lookUp):

Add a new helper for `-lookUpURL:isMainFrame:hasHighConfidenceOfSafety:completionHandler:`.

(WebKit::SafeBrowsingUtilities::listsForNamespace):

Add a new helper for `-getListsForNamespace:completionHandler:`. We&apos;ll make use of this in a
subsequent patch.

* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::beginSafeBrowsingCheck):

Adopt `WebKit::SafeBrowsingUtilities::(canLookUp|lookUp)`.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/303717@main">https://commits.webkit.org/303717@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1876e837104bb9024dd8de87f501be40474bc384

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133372 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5873 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44508 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140927 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/85419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6386 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5738 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102026 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/85419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/fdac7c43-88eb-4118-948a-b0633a728c69) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136319 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4535 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119547 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82850 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/16ca6b0a-2de7-4cc8-af14-15b94d434340) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/4414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/2003 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113510 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143574 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5543 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38241 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110401 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5625 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4760 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110580 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28030 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4278 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115802 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/59293 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5598 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34139 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5444 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/69050 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5687 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5554 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/Cocoa/SafeBrowsingUtilities.h:
```diff
@@ -23,16 +23,28 @@
-SOFT_LINK_CLASS_FOR_SOURCE(WebKit, SafariSafeBrowsing, SSBLookupResult);
-SOFT_LINK_CLASS_FOR_SOURCE(WebKit, SafariSafeBrowsing, SSBServiceLookupResult);
+namespace WebKit {
+namespace SafeBrowsingUtilities {
```
nit: nest them

```suggestion
namespace WebKit::SafeBrowsingUtilities {
```

whsieh replied:
Makes sense!

aprotyas commented on Source/WebKit/UIProcess/Cocoa/SafeBrowsingUtilities.mm:
```diff
@@ -0,0 +1,83 @@
+SOFT_LINK_CLASS_FOR_SOURCE(WebKit, SafariSafeBrowsing, SSBServiceLookupResult);
+
+namespace WebKit {
+namespace SafeBrowsingUtilities {
```
Same comment here if you want to take https://github.com/WebKit/WebKit/pull/54638/files#r2578126137

aprotyas commented on Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
```diff
@@ -268,38 +268,33 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
-            if (error)
-                return;
+    auto navigationType = forMainFrameNavigation ? SafeBrowsingUtilities::NavigationType::MainFrame : SafeBrowsingUtilities::NavigationType::SubFrame;
+    SafeBrowsingUtilities::lookUp(url, navigationType, [weakThis = WeakPtr { *this }, navigation = Ref { navigation }, forMainFrameNavigation, url = url.isolatedCopy(), redirectChainIndex] (SSBLookupResult *result, NSError *error) mutable {
```
style guide nit
```suggestion
    SafeBrowsingUtilities::lookUp(url, navigationType, [weakThis = WeakPtr { *this }, navigation = Ref { navigation }, forMainFrameNavigation, url = url.isolatedCopy(), redirectChainIndex](SSBLookupResult *result, NSError *error) mutable {
```

whsieh replied:
Fixed — thanks for the review!

whsieh committed:
[Safe Browsing] Consolidate SSBLookupContext usage in a single utility file
https://bugs.webkit.org/show_bug.cgi?id=303316
rdar://165620397

Reviewed by Abrar Rahman Protyasha.

Wrap code to call into SafariSafeBrowsing for lookup in C++ helper functions, in a new utility file
and namespace (`SafeBrowsingUtilities::lookUp`, etc.). No change in behavior.

* Source/WebKit/Platform/spi/Cocoa/SafeBrowsingSPI.h:
* Source/WebKit/SourcesCocoa.txt:

Add a new file, `SafeBrowsingUtilities.{h|mm}`, and move soft-linking to the source, since we now
only need to call directly into SSB classes in this file.

* Source/WebKit/UIProcess/Cocoa/BrowsingWarningCocoa.mm:
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.h: Removed.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingUtilities.h: Renamed from Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.mm.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingUtilities.mm: Added.
(WebKit::SafeBrowsingUtilities::canLookUp):
(WebKit::SafeBrowsingUtilities::lookUp):

Add a new helper for `-lookUpURL:isMainFrame:hasHighConfidenceOfSafety:completionHandler:`.

(WebKit::SafeBrowsingUtilities::listsForNamespace):

Add a new helper for `-getListsForNamespace:completionHandler:`. We'll make use of this in a
subsequent patch.

* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::beginSafeBrowsingCheck):

Adopt `WebKit::SafeBrowsingUtilities::(canLookUp|lookUp)`.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/303717@main
